cmd/gctcli/trades.go

```diff
@@ -0,0 +1,756 @@
+					Destination: &endTime,
+				},
+			},
+		},
+		{
+			Name:      "convertsavedtradestocandles",
+			Usage:     "explicitly converts stored trade data to candles and saves the result to the database",
+			ArgsUsage: "<exchange> <pair> <asset> <interval> <start> <end>",
+			Action:    convertSavedTradesToCandles,
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:  "exchange, e",
+					Usage: "the exchange",
+				},
+				cli.StringFlag{
+					Name:  "pair, p",
+					Usage: "the currency pair to get the trades for",
+				},
+				cli.StringFlag{
+					Name:  "asset, a",
+					Usage: "the asset type of the currency pair",
+				},
+				cli.Int64Flag{
+					Name:        "interval, i",
+					Usage:       fmt.Sprintf(klineMessage, "interval"),
+					Value:       86400,
+					Destination: &candleGranularity,
+				},
+				cli.StringFlag{
+					Name:        "start",
+					Usage:       "<start>",
+					Value:       time.Now().AddDate(0, -1, 0).Format(common.SimpleTimeFormat),
+					Destination: &startTime,
+				},
+				cli.StringFlag{
+					Name:        "end",
+					Usage:       "<end>",
+					Value:       time.Now().Format(common.SimpleTimeFormat),
+					Destination: &endTime,
+				},
+				cli.BoolFlag{
+					Name:  "sync, s",
+					Usage: "will sync the resulting candles to the database <true/false>",
+				},
+				cli.BoolFlag{
+					Name:  "force, f",
+					Usage: "will overwrite any conflicting candle data on save <true/false>",
+				},
+			},
+		},
+	},
+}
+
+func findMissingSavedTradeIntervals(c *cli.Context) error {
+	if c.NArg() == 0 && c.NumFlags() == 0 {
+		return cli.ShowCommandHelp(c, "findmissingsavedtradeintervals")
+	}
+
+	var exchangeName string
+	if c.IsSet("exchange") {
+		exchangeName = c.String("exchange")
+	} else {
+		exchangeName = c.Args().First()
+	}
+	if !validExchange(exchangeName) {
+		return errInvalidExchange
+	}
+
+	var currencyPair string
+	if c.IsSet("pair") {
+		currencyPair = c.String("pair")
+	} else {
+		currencyPair = c.Args().Get(1)
+	}
+	if !validPair(currencyPair) {
+		return errInvalidPair
+	}
+
+	p, err := currency.NewPairDelimiter(currencyPair, pairDelimiter)
+	if err != nil {
+		return err
+	}
+
+	var assetType string
+	if c.IsSet("asset") {
+		assetType = c.String("asset")
+	} else {
+		assetType = c.Args().Get(2)
+	}
+
+	if !validAsset(assetType) {
+		return errInvalidAsset
+	}
+
+	if !c.IsSet("start") {
+		if c.Args().Get(3) != "" {
+			startTime = c.Args().Get(3)
+		}
+	}
+
+	if !c.IsSet("end") {
+		if c.Args().Get(4) != "" {
+			endTime = c.Args().Get(4)
+		}
+	}
+
+	conn, err := setupClient()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err = conn.Close()
+		if err != nil {
+			fmt.Print(err)
+		}
+	}()
+
+	s, err := time.Parse(common.SimpleTimeFormat, startTime)
+	if err != nil {
+		return fmt.Errorf("invalid time format for start: %v", err)
+	}
+	e, err := time.Parse(common.SimpleTimeFormat, endTime)
+	if err != nil {
+		return fmt.Errorf("invalid time format for end: %v", err)
+	}
+
```

Comment:
If we move this check up before setupClient() on 255 it will save attempting to open a connection to the grpc server if timestamp parsing fails

Reply:
:+1: Just did this for all instances I came across for all CLI commands